app/realtime.php

```diff
@@ -449,15 +386,16 @@ function getCache(): Cache
         if (
             array_key_exists('realtime', $project->getAttribute('apis', []))
             && !$project->getAttribute('apis', [])['realtime']
-            && !(Auth::isPrivilegedUser(Authorization::getRoles()) || Auth::isAppUser(Authorization::getRoles()))
+            && !(Auth::isPrivilegedUser($authorization->getRoles()) || Auth::isAppUser($authorization->getRoles()))
         ) {
             throw new AppwriteException(AppwriteException::GENERAL_API_DISABLED);
         }
 
-        $dbForProject = getProjectDB($project);
-        $console = $app->getResource('console'); /** @var Document $console */
-        $user = $app->getResource('user'); /** @var Document $user */
-
+        $dbForProject = $container->get('getProjectDB')($project);
+        /** @var Document $console */
+        $console = $container->get('console');
+        /** @var Document $user */
+        $user = $container->refresh('user')->get('user');
```

Comment:
does `refresh()` make a new resource by making a new db query?

Reply:
`refresh()` yes, it will force the re-execution of this 👇 function
https://github.com/appwrite/appwrite/blob/1581a8c654e17ab1f44673866f30c62a76157e8e/app/init.php#L553-L565